CMakeLists.txt

```diff
@@ -77,9 +77,9 @@ if(MSVC)
   endif()
 else(MSVC)
   include(CheckCXXCompilerFlag)
-  check_cxx_compiler_flag("-std=c++11"    SUPPORT_CXX11)
+  check_cxx_compiler_flag("-std=c++14"    SUPPORT_CXX14)
```

Comment:
@BarclayII I'm using the latest version of cuda-gcc (9.3.0) so I must set standard to c++14 to make it work.
Could we make it a variable depending on the compiler version?

Reply:
This crashes on CentOS 7 where GCC 4.8.5 doesn't support C++14.

Does CUDA-GCC 9.3.0 really require C++14?  What was the error message it gave you if you use C++11?  My Fedora is using CUDA-GCC 8.3.0 and it worked fine with C++11 albeit without CUDA 11.

Reply:
Sorry, I have confirmed that C++14 is not necessarily required but thrust throws a warning encouraging me to compile DGL with C++11.
```
/opt/cuda/include/thrust/detail/config/cpp_dialect.h:104:13: warning: Thrust requires C++14. Please pass -std=c++14 to your compiler. Define THRUST_IGNORE_DEPRECATED_CPP_DIALECT to suppress this message.
```

Reply:
I see.  So thrust requires C++14.  You can switch the flag by checking CMake Variable `CUDA_VERSION_MAJOR`.